src/misc.rs

```diff
@@ -412,3 +553,38 @@ fn test_evaluate_index_poly() {
     let expected = evaluate_le(&index_polynomial, &x);
     assert_eq!(got, expected);
 }
+
+#[test]
+fn test_multi_poly_decompose() {
+    use crate::ark_std::Zero;
+    use ark_bls12_381::Fr;
+    let polynomial_small = vec![Fr::from(2_i32), Fr::from(3_i32)];
+    let eval_point_small: Vec<Fr> = vec![Fr::from(6_i32)];
+    let (quotients_small, remainder_small) =
+        multi_poly_decompose(&polynomial_small, &eval_point_small);
+    println!("testing small values");
+    assert_eq!(quotients_small[0][0], Fr::from(3_i32));
+    assert_eq!(remainder_small, Fr::from(20_i32));
+    println!("testing random values");
+    let reps = 3;
+    let dim = 12;
+    let rng = &mut ark_std::test_rng();
+    let polynomial: Vec<Fr> = random_vector(1 << dim, rng);
+    let eval_point: Vec<Fr> = random_vector(dim, rng);
```

Comment:
The variable `dim` sometimes refers the length of the vector, sometimes to the number of variables. this inconsistency is confusing

Reply:
I think we discussed this with Yuncong and decided that `dim` makes the most sense, since it is the number of variables. Whereas `deg` implies variables raised to powers greater than 1. Does that sound right?